### ISSUE TITLE
Where to find ASTConsumer and RecursiveASTVisitor of LLVM?

### ISSUE THREAD

**Issue by coder-chenzhi** [bytedeco/javacpp-presets#594]
I am learning to write a static checker by using LLVM. And I intend to follow [this tutorial](https://github.com/coder-chenzhi/clang-llvm-tutorial/blob/master/ast-interpreter/ASTInterpreter.cpp) to implement a ASTVisitor first. However, I didn't find ASTConsumer and RecursiveASTVisitor in JavaDoc of presets binding of LLVM.

**Comment by saudet** [comment]
`ASTConsumer` and `RecursiveASTVisitor` are not part of the C API. We would first need to create bindings for the C++ API to be able to use those.

**Comment by coder-chenzhi** [comment]
Do you have any plan to create bindings for the C++ API of LLVM?

**Comment by saudet** [comment]
Yes, as soon as we're done with https://github.com/bytedeco/javacpp/issues/51. Contributions welcome!

**Comment by saudet** [comment]
BTW, @Neiko2002 has contributed some samples for Clang using the C API (aka libclang):
https://github.com/bytedeco/javacpp-presets/tree/master/llvm/samples
If this API has all the features you need, that would be another option available to you.
